react-components/src/components/CacheProvider/RevisionFdmNodeCache.ts

```diff
@@ -245,7 +243,7 @@ export class RevisionFdmNodeCache {
     nodes: Node3D[]
   ): Array<{ connection: FdmCadConnection; treeIndex: TreeIndex }> {
     return mappingConnections.reduce((acc, connection) => {
-      const nodeInConnection = nodes.find((node) => node.id === connection.nodeId);
+      const nodeInConnection = nodes.find((node) => node.treeIndex === connection.treeIndex);
```

Comment:
As this mappingConnection is potentially large and we are more considering performance changes going forward, I would suggest to replace the `reduce ` to some regular for loop: default for loop, for-of, optimized-for, forEach or while.

Reply:
Right... Do you think that would reduce the time much? In any case, this function in particular is used only by the ancestor-mapping-finding function, and we shouldn't have more than about 20 results for that query normally

Reply:
Hmmm ok. I thought we could have a large array here. 
In general, using regular for-loop, for-of, forEach reduce considerably if the array is large (the "reduce" method is faster than for-in usually, but it is slower than the others mostly). You can make some benchmark to compare the average.